app/routes/_layout+/calendar.tsx

```diff
@@ -0,0 +1,229 @@
+import { useState, useRef, useCallback } from "react";
+import dayGridPlugin from "@fullcalendar/daygrid";
+import FullCalendar from "@fullcalendar/react";
+import { ChevronLeftIcon, ChevronRightIcon } from "@radix-ui/react-icons";
+import type { LoaderFunctionArgs, MetaFunction } from "@remix-run/node";
+import { json } from "@remix-run/node";
+import { Link, useLoaderData } from "@remix-run/react";
+import { ClientOnly } from "remix-utils/client-only";
+import FallbackLoading from "~/components/dashboard/fallback-loading";
+import Header from "~/components/layout/header";
+import { Button } from "~/components/shared/button";
+import { ButtonGroup } from "~/components/shared/button-group";
+import { Spinner } from "~/components/shared/spinner";
+import calendarStyles from "~/styles/layout/calendar.css?url";
+import { appendToMetaTitle } from "~/utils/append-to-meta-title";
+import { makeShelfError } from "~/utils/error";
+import { data, error } from "~/utils/http.server";
+import {
+  PermissionAction,
+  PermissionEntity,
+} from "~/utils/permissions/permission.validator.server";
+import { requirePermission } from "~/utils/roles.server";
+
+export function links() {
+  return [{ rel: "stylesheet", href: calendarStyles }];
+}
+
+export const handle = {
+  breadcrumb: () => <Link to="/calendar">Calendar</Link>,
+};
+
+// Loader Function to Return Bookings Data
+export const loader = async ({ request, context }: LoaderFunctionArgs) => {
+  const authSession = context.getSession();
+  const { userId } = authSession;
+
+  try {
+    // @TODO here we have to handle self-service, and make sure they can only see bookings that belong to them
```

Comment:
PLease remove this comment. This is solved already.

Reply:
Done